commit da60bfe8804af4d2d9578dd4479fd9e844956012 (refs/changes/19/3719/2)
Author: Bryan Cantrill <bryan@joyent.com>
Date:   2018-03-24T04:22:41+00:00 (1 year, 7 months ago)
    
    OS-6798 viona_recv_merged() can corrupt used ring
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Approved by: Patrick Mooney <patrick.mooney@joyent.com>

diff --git a/usr/src/uts/i86pc/io/viona/viona.c b/usr/src/uts/i86pc/io/viona/viona.c
index 359f814265..db5f46ae75 100644
--- a/usr/src/uts/i86pc/io/viona/viona.c
+++ b/usr/src/uts/i86pc/io/viona/viona.c
@@ -1670,6 +1670,7 @@ viona_recv_merged(viona_vring_t *ring, const mblk_t *mp, size_t msz)
 				 * drop the packet.
 				 */
 				err = EMSGSIZE;
+				break;
 			}
 			chunk = 0;
 			i = 0;
